File: SaneScanner/UIActivityViewController+SY.m
//
//  UIActivityViewController+SY.m
//  SaneScanner
//
//  Created by Stanislas Chevallier on 19/04/2017.
//  Copyright © 2017 Syan. All rights reserved.
//

#import "UIActivityViewController+SY.h"
#import "SYPDFActivity.h"
#import "SVProgressHUD+SY.h"

@implementation UIActivityViewController (SY)

+ (void)sy_showForUrls:(NSArray <NSURL *> *)urls
     fromBarButtonItem:(UIBarButtonItem *)barButtonItem
          presentingVC:(UIViewController *)presentingVC
            completion:(void(^)(void))completion
{
    SYPDFActivity *pdfActivity = [SYPDFActivity new];
    
    UIActivityViewController *activityViewController =
    [[UIActivityViewController alloc] initWithActivityItems:urls
                                      applicationActivities:@[pdfActivity]];
    
    [activityViewController setCompletionWithItemsHandler:^(UIActivityType activityType, BOOL completed, NSArray *returnedItems, NSError *activityError) {
        if (activityError)
            [SVProgressHUD showErrorWithStatus:activityError.sy_alertMessage];
        
        if (completion)
            completion();
    }];
    
    UIPopoverPresentationController *popover = activityViewController.popoverPresentationController;
    [popover setBarButtonItem:barButtonItem];
    
    pdfActivity.presentingViewController = presentingVC;
    [pdfActivity configureWithOriginalActivityViewController:activityViewController];

    [presentingVC presentViewController:activityViewController animated:YES completion:nil];
}

+ (void)sy_showForUrls:(NSArray <NSURL *> *)urls
  bottomInPresentingVC:(UIViewController *)presentingVC
            completion:(void(^)(void))completion
{
    SYPDFActivity *pdfActivity = [SYPDFActivity new];
    
    UIActivityViewController *activityViewController =
    [[UIActivityViewController alloc] initWithActivityItems:urls
                                      applicationActivities:@[pdfActivity]];
    
    [activityViewController setCompletionWithItemsHandler:^(UIActivityType activityType, BOOL completed, NSArray *returnedItems, NSError *activityError) {
        if (activityError)
            [SVProgressHUD showErrorWithStatus:activityError.sy_alertMessage];
        
        if (completion)
            completion();
    }];
    
    UIPopoverPresentationController *popover = activityViewController.popoverPresentationController;
    [popover setPermittedArrowDirections:0];
    [popover setSourceView:presentingVC.view];
    [popover setSourceRect:CGRectMake(presentingVC.view.bounds.size.width / 2.,
                                      presentingVC.view.bounds.size.height,
                                      1, 1)];
    
    pdfActivity.presentingViewController = presentingVC;
    [pdfActivity configureWithOriginalActivityViewController:activityViewController];
    
    [presentingVC presentViewController:activityViewController animated:YES completion:nil];
}

@end

File: SaneScanner/UIActivityViewController+SY.swift
//
//  UIActivityViewController+SY.swift
//  SaneScanner
//
//  Created by Stanislas Chevallier on 07/02/2019.
//  Copyright © 2019 Syan. All rights reserved.
//

import UIKit

#if !targetEnvironment(macCatalyst)
import SVProgressHUD
#endif


extension UIActivityViewController {

    static func showForURLs(_ urls: [URL], from barButtonItem: UIBarButtonItem, presentingVC: UIViewController, completion: (() -> ())?) {
        let pdfActivity = PDFActivity()
        
        let vc = UIActivityViewController(activityItems: urls, applicationActivities: [pdfActivity])
        
        vc.completionWithItemsHandler = { activityType, completed, returnedItems, error in
            if let error = error {
                SVProgressHUD.showError(withStatus: error.localizedDescription)
            }
            else if completed {
                completion?()
            }
        }
        
        vc.popoverPresentationController?.barButtonItem = barButtonItem
        pdfActivity.presentingViewController = presentingVC
        pdfActivity.configure(using: vc)
        
        presentingVC.present(vc, animated: true, completion: nil)
    }


    static func showForURLs(_ urls: [URL], fromBottomIn presentingVC: UIViewController, completion: (() -> ())?) {
        let pdfActivity = PDFActivity()
        
        let vc = UIActivityViewController(activityItems: urls, applicationActivities: [pdfActivity])
        
        vc.completionWithItemsHandler = { activityType, completed, returnedItems, error in
            if let error = error {
                SVProgressHUD.showError(withStatus: error.localizedDescription)
            }
            else if completed {
                completion?()
            }
        }
        
        vc.popoverPresentationController?.permittedArrowDirections = .up
        vc.popoverPresentationController?.sourceView = presentingVC.view
        vc.popoverPresentationController?.sourceRect = CGRect(x: presentingVC.view.bounds.width / 2, y: presentingVC.view.bounds.height, width: 1, height: 1)
        pdfActivity.presentingViewController = presentingVC
        pdfActivity.configure(using: vc)
        
        presentingVC.present(vc, animated: true, completion: nil)
    }
}
